src/vs/workbench/common/editor.ts

```diff
@@ -557,7 +564,7 @@ export interface IEditorInput extends IDisposable {
 	/**
 	 * Returns if the other object matches this input.
 	 */
-	matches(other: unknown): boolean;
+	matches(other: IEditorInput | IResourceEditorInputType): boolean;
```

Comment:
I think we should move `IResourceEditorInputType` into here now given it becomes such a core type of editors.